src/board/novacustom/v540tnx/gpio.c

```diff
@@ -0,0 +1,266 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+#include <board/gpio.h>
+#include <common/macro.h>
+
+// clang-format off
+struct Gpio __code ACIN_N =         GPIO(B, 0);
+struct Gpio __code AC_PRESENT =     GPIO(F, 7);
+struct Gpio __code ALL_SYS_PWRGD =  GPIO(C, 0);
+struct Gpio __code BKL_EN =         GPIO(C, 7);
+struct Gpio __code BUF_PLT_RST_N =  GPIO(D, 2);
+struct Gpio __code CCD_EN =         GPIO(D, 1);
+struct Gpio __code DD_ON =          GPIO(E, 4);
+struct Gpio __code DGPU_PWR_EN =    GPIO(H, 4);
+struct Gpio __code EC_EN =          GPIO(B, 6);
+struct Gpio __code EC_RSMRST_N =    GPIO(E, 5);
+struct Gpio __code GC6_FB_EN =      GPIO(J, 3);
+struct Gpio __code H_PROCHOT_EC =   GPIO(G, 6);
+struct Gpio __code JACK_IN_N =      GPIO(C, 6);
+struct Gpio __code LAN_WAKEUP_N =   GPIO(B, 2);
+struct Gpio __code LED_ACIN =       GPIO(H, 2);
+struct Gpio __code LED_BAT_CHG =    GPIO(H, 5);
+struct Gpio __code LED_BAT_FULL =   GPIO(J, 0);
+struct Gpio __code LED_PWR =        GPIO(D, 0);
+struct Gpio __code LID_SW_N =       GPIO(B, 1);
+struct Gpio __code PCH_PWROK_EC =   GPIO(F, 3);
+struct Gpio __code PD_EN =          GPIO(D, 4);
+struct Gpio __code PWR_BTN_N =      GPIO(D, 5);
+struct Gpio __code PWR_SW_N =       GPIO(B, 3);
+struct Gpio __code RGBKB_DET_N =    GPIO(I, 2);
+struct Gpio __code SINK_CTRL =      GPIO(H, 7);
+struct Gpio __code SUSB_N_PCH =     GPIO(H, 0);
+struct Gpio __code SUSC_N_PCH =     GPIO(H, 1);
+struct Gpio __code VA_EC_EN =       GPIO(J, 4);
+struct Gpio __code WLAN_PWR_EN =    GPIO(E, 1);
+struct Gpio __code XLP_OUT =        GPIO(B, 4);
+// clang-format on
+
+void gpio_init(void) {
+    // Set global configuration
+    // Enable LPC reset on GPD2
+    GCR = 0b10 << 1;
+    // Disable UARTs
+    GCR6 = 0;
+    // PWRSW WDT 2 Enable 1
+    GCR8 = BIT(4);
+    // PWRSW WDT 2 Enable 2
+    GCR9 = BIT(5);
+    // Enable SMBus channel 4
+    GCR15 = BIT(4);
+    // Set GPB5 and GPD2 to 1.8V
+    GCR19 = BIT(7) | BIT(0);
+    // Set GPD3 to 1.8V, GPF2 and GPF3 to 3.3V
+    GCR20 = BIT(7);
+    // Set GPF6, GPF7, GPH0, and GPH1 to 1.8V
+    GCR21 = BIT(6), BIT(5) | BIT(2) | BIT(1);
+    // Set VCC power domain to 1.8V
+    GCR22 = BIT(7);
+    // Set GPM6 power domain to VCC
+    GCR23 = BIT(0);
+
+    // Set GPIO data
+    GPDRA = 0;
+    // XLP_OUT, PWR_SW#
+    GPDRB = BIT(4) | BIT(3);
+    GPDRC = 0;
+    // PWR_BTN#
+    GPDRD = BIT(5);
+    // USB_PWR_EN
+    GPDRE = BIT(3);
+    // H_PECI
+    GPDRF = BIT(6);
+    // H_PROCHOT_EC
+    GPDRG = BIT(6);
+    // BL_PWM_EN_EC, PLVDD_RST_EC
+    GPDRH = BIT(6) | BIT(3);
+    GPDRI = 0;
+    // KBC_MUTE#
+    GPDRJ = BIT(1);
+    GPDRM = 0;
+
+    // Set GPIO control
+
+    // EC_PWM_LEDKB_P
+    GPCRA0 = GPIO_ALT;
+    // KBC_BEEP
+    GPCRA1 = GPIO_IN;
```

Comment:
KBC_BEEP should be OUT?

Reply:
fixed by https://github.com/Dasharo/ec/pull/62/commits/d3744e11f63907f29d0eb2e4b65ad3c54206d48e (although we don't use this feature)